### PR TITLE
feat: add standalone guide on XLA support.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

We have had XLA support for our TF generation models (GPT2, Whisper, for example) for a while. This PR adds a standalone guide in the doc to discuss it. 

Cc: @Rocketknight1 @amyeroberts @ydshieh

### CODE REVIEW

**Comment by ydshieh** [comment]
@sayakpaul 

It seems there is an issue with your CircleCI permissions.
Could you try refreshing your permissions as shown [here](https://support.circleci.com/hc/en-us/articles/360048210711-How-to-Refresh-User-Permissions-)?

**Comment by sayakpaul** [comment]
@ydshieh I don't have CircleCI signed in nor have I installed it here: https://github.com/settings/applications.

**Comment by ydshieh** [comment]
Could you login CircleCI with your GitHub account, and follow `huggingface/transformers`?

See https://circleci.com/docs/projects/

(but it's kinda strange - you have opened a lot of PRs before, so not sure why we have this issue now)

**Comment by sayakpaul** [comment]
Should be all good now I guess:

image

**Review by gante — APPROVED** [review]
(As pre-approved in the google docs draft :) )

File: docs/source/en/tf_xla.mdx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+# Generate random inputs for the model.
+batch_size = 16
+input_vector_dim = 10
+random_inputs = tf.random.normal((batch_size, input_vector_dim))
+
+# Run a forward pass.
+_ = model(random_inputs)
+```
+
+In order to run the forward pass with an XLA-compiled function, we’d need to:
```
Comment (sgugger) [inline docs/source/en/tf_xla.mdx:49]:
```suggestion
In order to run the forward pass with an XLA-compiled function, we’d need to do:
```

File: docs/source/en/tf_xla.mdx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+The default `call()` function of the `model` is used for compiling the XLA graph. But if there’s any other model function you want to compile into XLA that’s also possible with:
+
+```py
+my_xla_fn = tf.function(model.my_xla_fn, jit_compile=True)
+```
+
+## Running a TF text generation model with XLA from 🤗 Transformers
+
+To enable XLA-accelerated generation within 🤗 Transformers, you need to have the correct version of `transformers` installed. You can install it by running:
```
Comment (sgugger) [inline docs/source/en/tf_xla.mdx:64]:
```suggestion
To enable XLA-accelerated generation within 🤗 Transformers, you need to have a recent version of `transformers` installed. You can install it by running:
```
(the suggests there is only one)

File: docs/source/en/tf_xla.mdx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+```bash
+pip install transformers>=4.21.0
```
Comment (sgugger) [inline docs/source/en/tf_xla.mdx:67]:
```suggestion
pip install transformers --upgrade
```
We want users to be on the latest version.

File: docs/source/en/tf_xla.mdx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+```
+
```
Comment (sgugger) [inline docs/source/en/tf_xla.mdx:69]: Maybe add a code sample that displays the version and instruct the user to check it is greater than 4.21.0?

**Review by sgugger — APPROVED** [review]
Thanks for this new guide. You should make it an auto-generated notebook by adding `[[open-in-colab]]` at the top.

**Comment by sayakpaul** [comment]
@sgugger addressed your comments including [this one](https://github.com/huggingface/transformers/pull/21141/files#r1072172489). Let me know if I am good to merge given all tests are green.

**Comment by sgugger** [comment]
Yes, all good!

**Pull request merged**